src/ripple/ledger/impl/PaymentSandbox.cpp

```diff
@@ -172,7 +172,11 @@ PaymentSandbox::balanceHook (AccountID const& account,
         }
     }
 
-    assert (!isXRP(issuer) || adjustedAmt >= beast::zero);
+    if (isXRP(issuer) && adjustedAmt < beast::zero)
+    {
+        adjustedAmt.clear();
```

Comment:
Yeah, it's nothing to worry about. Consider a payment snippet that credits a large XRP amount and then debits the same amount. By our rules, we can't use the credit but we subtract the debit, so we'll calculate a negative value. It's not an error case.